src/nl/surf/eduhub_rio_mapper/api.clj

```diff
@@ -14,16 +16,31 @@
             [ring.middleware.json :refer [wrap-json-response]])
   (:import java.util.UUID))
 
+(defn extract-resource [uri]
+  (when uri
+    (if (str/starts-with? uri "/job/")
+      (-> uri (subs 5) (str/split #"/" 2) last)             ; Remove action
+      (log/error (str "invalid uri: " uri)))))
+
 (defn wrap-job-enqueuer
   [app enqueue-fn]
-  (fn with-job-enqueuer [req]
+  (fn with-job-enqueuer [{:keys [callback] :as req}]
     (let [{:keys [job] :as res} (app req)]
       (if job
-        (let [token (UUID/randomUUID)]
-          (enqueue-fn (assoc job :token token))
-          (assoc res :body {:token token}))
+        (let [attrs {:token    (UUID/randomUUID)
+                     :callback callback
+                     :resource (extract-resource (:uri req))}]
+          (enqueue-fn (into {} (filter val (merge job attrs))))
+          (assoc res :body attrs))
         res))))
 
+(defn wrap-callback-extractor [app]
+  (fn callback-extractor [req]
+    (let [callback (get-in req [:headers "x-callback"])]
+      (app (if callback
+             (assoc req :callback callback)
```

Comment:
It's an URL so name it `callback-url` too.

Also, maybe this is a better place to do `extract-resource`?